### PR TITLE
Feat: add ReduceLROnPlateau support

### LINKED ISSUE #160: Loss dependent scheduler not supported (e.g. ReduceLROnPlateau)
**Describe the bug**

When using ReduceLROnPlateau scheduler, the scheduler expects input of epoch loss value to identify Plateau and raises error.
Adding stopping_loss as parameter to scheduler.step() in the code also does not help since it is located in the training section 
and not after the validation evaluation section.
I recommend as solution to move the scheduler.step lines to the Class TabModel section in the def of fit (line 244) and include stopping_loss (this time it is valid stopping_loss) as parameter.

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**Expected behavior**

**Screenshots**

**Other relevant information:**
poetry version:  
python version: 
Operating System: 
Additional tools:

**Additional context**

### PR DESCRIPTION
**Pull request by bennyjg**
Enable use of lr_scheduler.ReduceLROnPlateau which requires a loss parameter during the step phase.
Change suggested - move scheduler.step() to outer training loop and add stopping_loss as parameter for case of ReduceLROnPlateau scheduler.

**What kind of change does this PR introduce?**
Enhancement

**Does this PR introduce a breaking change?**
Not to my best knowledge

**What needs to be documented once your changes are merged?**
indicate support for this scheduler in the documentation. This LR_scheduler can be very useful.

**Closing issues**
Closes #160

### CODE REVIEW

**Review by eduardocarvp — CHANGES_REQUESTED** [review]
Thanks, @bennyjg !
I see no problem removing the scheduler step from `train_epoch` function and adding it to the main loop if the loss is needed. However, I guess you should remove the same scheduler lines below from the `TabNetRegressor` class as well:
```
if self.scheduler is not None:
    self.scheduler.step()
```

**Comment by bennyjg** [comment]
True, I missed that part.
Should I add it as another commit?
Benny

**Comment by bennyjg** [comment]
> Thanks, @bennyjg !
> I see no problem removing the scheduler step from `train_epoch` function and adding it to the main loop if the loss is needed. However, I guess you should remove the same scheduler lines below from the `TabNetRegressor` class as well:
> 
> ```
> if self.scheduler is not None:
>     self.scheduler.step()
> ```

Hi @eduardocarvp , I made the additional change you suggested. Thanks!
I hope it is now ready to be merged. :)

**Comment by Optimox** [comment]
@bennyjg I think we are almost good! Any chance you could squash your last commit to have only one commit for this PR?

**Comment by bennyjg** [comment]
> @bennyjg I think we are almost good! Any chance you could squash your last commit to have only one commit for this PR?

Is there a Git trick to do that without resubmitting the whole CR from scratch?
Like a merge button of some sort??

**Comment by eduardocarvp** [comment]
@bennyjg, sure! You can use `git rebase -i HEAD~2` and then squash the second commit. Remember to `push --force-with-lease` after you're done.

Some doc here if you don't know `git rebase -i`:  https://opensource.com/article/20/4/git-rebase-i

**Comment by Optimox** [comment]
Well this will cost me some conflicts with my PR #184 but you were first! Thank you very much for contributing to pytorch-tabnet, your help is much appreciated!

**Review by Optimox — APPROVED** [review]
Thanks approved!

**Comment by bennyjg** [comment]
Thanks! Enjoyed it :)
Hope to see more exciting features added.
